### ISSUE TITLE
Feature request: Add TCP protocol to the auxiliary/scanner/snmp/snmp_login

### ISSUE THREAD

**Issue by rstewartr7** [rapid7/metasploit-framework#9649]
The auxiliary/scanner/snmp/snmp_login only checks UDP ports, and I would like to request the addition of a 'Protocol switch' to be able to check hosts listening on TCP.

### Metasploit version.
Framework: 4.16.40-dev-
Console  : 4.16.40-dev-

### I installed Metasploit with:
- [ ] Kali package via apt
- [X] Omnibus installer (nightly)
- [ ] Commercial/Community installer (from http://www.rapid7.com/products/metasploit/download.jsp)
- [ ] Source install (please specify ruby version)

### OS
Kali

**Comment by CaptainFreak** [comment]
This looks like a very good way to start contributing to this community.
I looked in to doing SNMP over tcp and its generally not done. [Link](https://stackoverflow.com/questions/3565975/why-is-snmp-usually-run-over-udp-and-not-tcp-ip)

But perhaps sometimes it can be used over TCP. Looked into snmp_login.rb but couldn't find a way to start. If you can, please point me some reference code which does the similar protocol switch.

**Comment by CaptainFreak** [comment]
PR : #10841

**Comment by nrathaus** [comment]
Trying to redo some of the missing work here...

Lets start with getting a testing environment (listen on TCP and UDP)
```
docker run -d --name snmpd -p 161:161/tcp -p 161:161/udp polinux/snmpd udp:161 tcp:161
```

Make sure it is (docker) working as expected
```
snmpwalk -Os -c public -v 1 tcp:127.0.0.1 .
```

**Comment by nrathaus** [comment]
Idea is to add an attribute to `/opt/metasploit-framework/embedded/framework/lib/metasploit/framework/login_scanner/snmp.rb` to pick whether UDP or TCP should be used

Then split the code so that the socket is either UDP or TCP created, and the subsequent functions are used respectively (recv, send, instead of recvfrom and sendto)

**Comment by nrathaus** [comment]
Added PR https://github.com/rapid7/metasploit-framework/pull/19064

**Comment by nrathaus** [comment]
Metasploit testing:
```
use auxiliary/scanner/snmp/snmp_login
set RHOSTS 127.0.0.1
set PROTOCOL tcp
run
```

**Comment by baggers27** [comment]
Looks like this has been merged under #19064 and can be closed?
